frontend/src/utils/api.js

```diff
@@ -4,16 +4,24 @@ import {
   purgeGlobalAuthState
 } from "./apiHelpers";
 
-async function getSearchResults(keyword, address) {
+async function getSearchResults(keyword, address, tag) {
   const defaultRadius = 500;
   let endptStr = `resources/filter?`;
 
-  if (address && keyword) {
+  if (address && keyword && tag) {
+    endptStr += `radius=${defaultRadius}&address=${address}&keyword=${keyword}&tag=${tag}`;
+  } else if (address && keyword) {
     endptStr += `radius=${defaultRadius}&address=${address}&keyword=${keyword}`;
+  } else if (address && tag) {
+    endptStr += `radius=${defaultRadius}&address=${address}&tag=${tag}`;
```

Comment:
let's do something like:
```
if (radius) {
    endptStr  += `radius=${defaultRadius}`
}
if (address) {
    endptStr += `&address=${address}&keyword=${keyword}`;
  }
if (keyword) {
    endptStr += `&keyword=${keyword}`;
  } 
if (tag) {
    endptStr += `&tag=${tag}`;
}
```

so we don't have to enumerate all possibilities

Reply:
wait but if there's no radius there would be an extra '&' at the beginning of the other fields

Reply:
radius is always set, this will just prep us for being able to pass in radius as a search param too

Reply:
We can actually just do this really cleanly using `urljoin` which is a module we're using for the Redux API middleware. You can import from `url-join` like in `apiHelpers.js` and then do something like [this](https://github.com/jfromaniello/url-join/blob/master/test/tests.js#L83). To be really fancy you can try using the [spread operator](https://medium.com/@luke_smaki/javascript-es6-spread-operator-and-rest-parameters-b3e89d112281) to put this in to parameters in the function.

Reply:
ah, that's what i was looking for!